test/framework/imageregistry.go

```diff
@@ -154,23 +155,88 @@ func deleteImageRegistryResource(te TestEnv) {
 				context.Background(), defaults.ImageRegistryResourceName, metav1.GetOptions{},
 			)
 		},
-		func(deleteOptions *metav1.DeleteOptions) error {
+		func(deleteOptions metav1.DeleteOptions) error {
 			return te.Client().Configs().Delete(
-				context.Background(), defaults.ImageRegistryResourceName, *deleteOptions,
+				context.Background(), defaults.ImageRegistryResourceName, deleteOptions,
 			)
 		},
-	); err != nil && !errors.IsNotFound(err) {
+	); err != nil {
 		te.Fatalf("unable to delete the image registry resource: %s", err)
 	}
 }
 
+func deleteNodeCADaemonSet(te TestEnv) {
+	ds, err := te.Client().DaemonSets(defaults.ImageRegistryOperatorNamespace).Get(
+		context.Background(),
+		"node-ca",
+		metav1.GetOptions{},
+	)
+	if errors.IsNotFound(err) {
+		return
+	}
+	if err != nil {
+		te.Fatalf("unable to get the current state of daemonset/node-ca: %s", err)
+	}
+
+	policy := metav1.DeletePropagationBackground
```

Comment:
Isn't it possible to use `DeletePropagationForeground` here so we don't need to wait with `WaitUntilFinalized()`?

Reply:
DeletePropagationForeground doesn't wait the object is finalized anyway, it's an instruction for GC. For daemonsets deleting its children can be slow. I'd prefer it to stay as a background job.